src/parser.ts

```diff
@@ -37,48 +44,90 @@ const stateSet = {
     [TOKEN_TYPES.closeBlock]: { match: ']', push: 'main' },
   },
 };
-export function parser(config: string) {
+export function tokenize(config: string, filePath: string, pathMemo = new Set()): FluentBitToken[] {
   if (!config.replace(/\s/g, '')) {
-    throw new Error('Invalid config file');
+    throw new TokenError('File is empty', filePath, 0, 0);
   }
 
   if (!isFluentBit(config)) {
-    throw new Error('This file is not a valid Fluent Bit config file');
+    throw new TokenError('This file is not a valid Fluent Bit config file', filePath, 0, 0);
   }
 
-  const lexer = states(stateSet);
-  const tokens = lexer.reset(config);
+  let tokens = [] as FluentBitToken[];
+  const lexer = states(stateSet).reset(config);
+
+  // We will expand every include first, looking for any missing paths and invalid tokens
+  // https://github.com/calyptia/fluent-bit-config-parser/issues/15
+  for (const token of lexer) {
+    if (token.type === TOKEN_TYPES.include) {
+      const [, includeFilePath] = token.value.split(' ');
```

Comment:
We probably should catch multiple splits here - the only valid option is `@include filename` and not `@include file name` etc. I imagine though this will be handled by the path stuff below or will that combine the `file` and `name` into a single valid path (which is wrong)?

Reply:
Possibly needs another test case just to verify and prevent any future issues.

Reply:
Fixed as well, with it's very own Error.